### ISSUE TITLE
[az ad]: Add sign-in-audience as enum arg to create and update

### ISSUE THREAD

**Issue by stevengum** [Azure/azure-cli#8820]
**Is your feature request related to a problem? Please describe.**
`az ad app create` and `az ad app update` do not support `signInAudience` which means a user who wishes to change this value must go to the portal to change the setting.

**Describe the solution you'd like**
`az ad app create` and `az ad app update` to support --sign-in-audience as an enum type.

**Additional Context**
The currently published azure-graphrbac SDK does not have the correct values for [signInAudience](https://github.com/Azure/azure-sdk-for-python/blob/62d94260a04c5764ac8dbb5f8853f65f1ea9be17/azure-graphrbac/azure/graphrbac/models/application_create_parameters.py#L106-L109).

This is blocked by Azure/azure-sdk-for-python#4574 (or another PR with the _correct_ values) which is in turn blocked by Azure/azure-rest-api-specs#5427

**Comment by yonzhan** [comment]
@jiasli please take a look and comment.

**Comment by jiasli** [comment]
This is blocked by Azure/azure-rest-api-specs#5427

**Comment by yonzhan** [comment]
@darshanhs90 this is blocked by Azure/azure-rest-api-specs#5427, please take a look.
